server/src/slice_config.rs

```diff
@@ -1,45 +1,62 @@
 // Copyright (c) ZeroC, Inc.
 
-use std::path::{Path, PathBuf};
+use std::path::Path;
 
-// This struct holds the configuration for a single compilation set.
+use slicec::slice_options::SliceOptions;
+
+/// This struct holds configuration that affects the entire server.
+#[derive(Debug, Default)]
+pub struct ServerConfig {
+    /// This is the root path of the workspace, used to resolve relative paths. It must be an absolute path.
+    pub workspace_root_path: String,
+    /// This is the path to the built-in Slice files that are included with the extension. It must be an absolute path.
+    pub built_in_slice_path: String,
+}
+
+/// This struct holds the configuration for a single compilation set.
 #[derive(Debug)]
 pub struct SliceConfig {
-    pub paths: Vec<String>,
-    pub workspace_root_path: Option<PathBuf>,
-    pub built_in_slice_path: Option<String>,
+    /// List of paths that will be passed to the compiler as reference files/directories.
+    pub slice_search_paths: Vec<String>,
+    /// Specifies whether to include the built-in Slice files that are bundled with the extension.
+    pub include_built_in_slice_files: bool,
 }
 
-impl SliceConfig {
-    // Resolve path URIs to file paths to be used by the Slice compiler.
-    pub fn resolve_paths(&self) -> Vec<String> {
-        // If `root_path` isn't set, relative path resolution is impossible, so we return.
-        let Some(root_path) = &self.workspace_root_path else {
-            return vec![];
-        };
+impl Default for SliceConfig {
+    fn default() -> Self {
+        SliceConfig {
+            slice_search_paths: vec![],
+            include_built_in_slice_files: true,
+        }
+    }
+}
 
-        let mut resolved_paths = Vec::new();
-        for string_path in &self.paths {
-            let path = Path::new(string_path);
+pub fn compute_slice_options(server_config: &ServerConfig, unit_config: &SliceConfig) -> SliceOptions {
```

Comment:
❓ Why is this called `unit_config`, seems like a bit of an awkward name, makes me think of 'default' or like the unit sphere etc if that makes sense. Both cases we call this should not involve the default `SliceConfig`

Reply:
I see where you're coming from. I renamed it to `set_config` (short for `configuration_set` (because `configuration_set_config` sounds strange)).

I had meant unit in the sense of a 'compilation _unit_' (a group of files that are compiled together in compiler jargon).
Just for reference.